org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/queues/QueueViews.java

```diff
@@ -1,149 +1,149 @@
+
+	/**
+	 *
+	 * @return
+	 */
+	public static final String getQueueViewID() {
+		return "org.eclipse.scanning.event.ui.queueView"; // Might need to move this to a preference or other property.
+	}
+
+	/**
+	 *
+	 * @param uri
+	 * @param bundle
+	 * @param className
+	 * @param partName
+	 * @return
+	 */
+	public static String createId(String uri, String bundle, String className, String partName) {
+		return createId(uri, bundle, className, EventConstants.STATUS_SET, EventConstants.STATUS_TOPIC, EventConstants.SUBMISSION_QUEUE, partName);
+	}
+
+	/**
+	 *
+	 * @param uri
+	 * @param bundle
+	 * @param bean
+	 * @param queueName
+	 * @param topicName
+	 * @param submissionQueueName
+	 * @param partName
+	 * @return
+	 */
+	public static String createId(String uri, String bundle, String bean,
+								final String queueName,
+								final String topicName,
+								final String submissionQueueName,
+								String partName)  {
+
+
+		String queueViewId = QueueViews.createSecondaryId(uri, bundle,bean, queueName, topicName, submissionQueueName);
+		if (partName!=null) queueViewId = queueViewId+"partName="+partName;
+
+		final StringBuilder buf = new StringBuilder();
+		buf.append(getQueueViewID());
+		buf.append(":");
+		buf.append(queueViewId);
+		return buf.toString();
+	}
+
+	/**
+	 *
+	 * @param beanBundleName
+	 * @param beanClassName
+	 * @param queueName
+	 * @param topicName
+	 * @param submissionQueueName
+	 * @return
+	 */
+	public static String createSecondaryId(final String beanBundleName, final String beanClassName, final String queueName, final String topicName, final String submissionQueueName) {
+        return createSecondaryId(null, beanBundleName, beanClassName, queueName, topicName, submissionQueueName);
+	}
+
+	/**
+	 *
+	 * @param uri
+	 * @param beanBundleName
+	 * @param beanClassName
+	 * @param queueName
+	 * @param topicName
+	 * @param submissionQueueName
+	 * @return
+	 */
+	public static String createSecondaryId(String uri, final String beanBundleName, final String beanClassName, final String queueName, final String topicName, final String submissionQueueName) {
+
+		final StringBuilder buf = new StringBuilder();
+		if (uri!=null) {
+			try {
+				uri = URLEncoder.encode(uri, "UTF-8");
+			} catch (UnsupportedEncodingException e) {
+				e.printStackTrace(); // Not fatal
+			}
+			append(buf, "uri",      uri);
+		}
+		append(buf, "beanBundleName",      beanBundleName);
+		append(buf, "beanClassName",       beanClassName);
+		append(buf, "queueName",           queueName);
+		append(buf, "topicName",           topicName);
+		append(buf, "submissionQueueName", submissionQueueName);
+		return buf.toString();
+	}
+
+
+	protected static String createSecondaryId(String uri, String requestName, String responseName) {
+		final StringBuilder buf = new StringBuilder();
+		if (uri!=null) {
+			try {
+				uri = URLEncoder.encode(uri, "UTF-8");
+			} catch (UnsupportedEncodingException e) {
+				e.printStackTrace(); // Not fatal
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)